### PR TITLE
Testplans for the Bugfinder

### PR DESCRIPTION
**Pull request by madebyTimo**
Closes Gamify-IT/issues#177

### CODE REVIEW

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Bugfinder next code (u.bugfinder-6)
+
+
+Version: V1.0, 05.08.2022
+Author: Timo Schnaible
+Tester: -
+
+## Description
+
+When the player submited his solution he can get to the next code snipped by pressing the `Next Code` button which replaced the `Submit` button. The button is only shown if there is another code snipped. Additionally the virtual chat partner asks for help with another bug in the chat.
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md:10]:
```suggestion
After submitting a solution, there can be a new snippet with new bugs. Additionally, the virtual chat partner asks for help again.
```

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Precondition
+
+The tester submited and another code snipped exists.
+
+## Postcondition
+
+The next code snipped is shown.
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md:18]:
```suggestion
The next code snippet is shown.
```
At least you're consistently wrong...

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Typical procedure
+
+1. Press start to start the game
+2. The first code snipped is shown
+3. Select the bugs
+4. Press the `Submit` button
+5. The `Submit` button gets replaced by the `Next Code` button
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md:26]: Precondition. Maybe apart from ``The `Submit` button is now a `Next Code` button`` here.

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+## Criticality
+
+High
+
+## Linkages
+
+- [Feedback (u.bugfinder-5)](u-bugfinder-4-feedback.md)
+- [Bug selection (u.bugfinder-2)](u-bugfinder-2-bug-selection.md)
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md:54]:
```suggestion
- [Feedback (`u.bugfinder-4`)](u-bugfinder-4-feedback.md)
- [Bug selection (`u.bugfinder-2`)](u-bugfinder-2-bug-selection.md)
```

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+5.1. The `Next Code` button doesn't appear
+5.2. The `Submit` button doesn't disappear
+6.1. No chat message appears
+6.2. A wrong chat message appears
+8.1. No chat message appears
+8.2. A wrong chat message appears
+10.1. No chat message appears
+10.2. The virtual chat partner is unhappy about the help
+11.1. Th next code snipped doesn't appear
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-next-code.md:44]:
```suggestion
11.1. The next code snippet doesn't appear
```

**Review by delvh — APPROVED** [review]
LGTM. (Probably because I rewrote it until I couldn't find a mistake anymore... )

**Comment by delvh** [comment]
Deadline for new reviews is tonight@23:00 so that the issue can be closed today. Or Burndown chart looks already bad enough in this sprint.

**Review by cdingm — DISMISSED** [review]

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# Bugfinder` Show general feedback (`u.bugfinder-4`)
```
Comment (maexled) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md:1]:
```suggestion
# Bugfinder Show general feedback (`u.bugfinder-4`)
```

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+
+Version: V1.1, 05.08.2022, Improve format and wording	
+Author: Timo Schnaible, Leon Hofmeister
+Tester: -
```
Comment (maexled) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md:6]:
```suggestion
Version: V1.1, 05.08.2022, Improve format and wording

Author: Timo Schnaible, Leon Hofmeister

Tester: -
```
When all is on one line, it makes it harder to read, especially when there are comments about the updated version

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-5-show-detailed-feedback.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+## Alternative procedures
+
+4.1 The popup doesn't appear \
+4.2 The popup is empty \
+4.3 The popup contains the wrong feedback \
+6.1 The popup doesn't disappear
+
+## Criticality
+
+Medium
```
Comment (maexled) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-5-show-detailed-feedback.md:38]: Why is this medium, but the chat feedback critical?

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-show-next-snippet.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+1.1. The `Next Code` button doesn't appear
+1.2. The `Submit` button doesn't disappear
+2.1. No chat message appears
+2.2. A wrong chat message appears
+2.2.1. The chat partner doesn't want any further help
+4.1. No chat message appears
+4.2. A wrong chat message appears
+5.1. No chat message appears
+5.2. The virtual chat partner is unhappy about the help
+6.1. The next code snippet doesn't appear
```
Comment (maexled) [inline dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-6-show-next-snippet.md:41]:
```suggestion
1.1. The `Next Code` button doesn't appear \
1.2. The `Submit` button doesn't disappear \
2.1. No chat message appears \
2.2. A wrong chat message appears \
2.2.1. The chat partner doesn't want any further help \
4.1. No chat message appears \
4.2. A wrong chat message appears \
5.1. No chat message appears \
5.2. The virtual chat partner is unhappy about the help \
6.1. The next code snippet doesn't appear
```
to be consistent

**Review by maexled — COMMENTED** [review]
Other things I have noticed:
cancel-bug-submission -> replace `cancel` to `undo` or `revert`, cancel is in my opinion when you have the modal open to specify a bug, but you hit cancel (we also have no test plan for this, I do not know if we need this)

Also on one point it is `bug-selection`, but the other `bug-submission`

File: dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md (hunk 7 above)
Comment (delvh) [reply dev-manuals/test-plans/minigames/bugfinder-frontend/plans/u-bugfinder-4-show-general-feedback.md:6]: Not intended to be one line.
Will add explicit line breaks.

**Comment by delvh** [comment]
> Also on one point it is bug-selection, but the other bug-submission

Are you referring to `u-2` and `u-3`?
If yes, that's because they talk about two separate actions, hence `selection` is more fitting for `u-2` while `submission` is more fitting for `u-3`.

**Review by delvh — APPROVED** [review]

**Review by maexled — APPROVED** [review]

**Comment by delvh** [comment]
Regardless, `submission` in `u.bugfinder-3` has now been changed to `correction`.

**Pull request merged**